client/network/src/request_responses.rs

```diff
@@ -279,28 +281,7 @@ pub struct RequestResponsesBehaviour {
 	send_feedback: HashMap<ProtocolRequestId, oneshot::Sender<()>>,
 
 	/// Primarily used to get a reputation of a node.
-	peerset: PeersetHandle,
-
-	/// Pending message request, holds `MessageRequest` as a Future state to poll it
-	/// until we get a response from `Peerset`
-	message_request: Option<MessageRequest>,
-}
-
-// This is a state of processing incoming request Message.
-// The main reason of this struct is to hold `get_peer_reputation` as a Future state.
-struct MessageRequest {
-	peer: PeerId,
-	request_id: RequestId,
-	request: Vec<u8>,
-	channel: ResponseChannel<Result<Vec<u8>, ()>>,
-	protocol: ProtocolName,
-	// A builder used for building responses for incoming requests. Note that we use
-	// `async_channel` and not `mpsc` on purpose, because `mpsc::channel` allocates an extra
-	// message slot for every cloned `Sender` and this breaks a back-pressure mechanism.
-	resp_builder: Option<async_channel::Sender<IncomingRequest>>,
-	// Once we get incoming request we save all params, create an async call to Peerset
-	// to get the reputation of the peer.
-	get_peer_reputation: Pin<Box<dyn Future<Output = Result<i32, ()>> + Send>>,
```

Comment:
This is very nice. I wanted to refactor this code because I didn't like the reputation future at all but you beat me to it.